cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureUtils.java

```diff
@@ -348,19 +352,54 @@ public void deleteManagedDisks(AzureClient azureClient, Collection<String> manag
     }
 
     @Retryable(backoff = @Backoff(delay = 1000, multiplier = 2, maxDelay = 10000), maxAttempts = 5)
-    public void deleteDatabaseServer(AzureClient azureClient, String databaseServerId) {
-        LOGGER.info("Delete database server: {}", databaseServerId);
-        List<String> failedToDeleteDatabaseServer = new ArrayList<>();
-        Completable databaseServerDeleteCompletable = azureClient.deleteDatabaseServer(databaseServerId)
-                .doOnError(throwable -> {
-                    LOGGER.error("Error happend on azure database server delete: {}", databaseServerId, throwable);
-                    failedToDeleteDatabaseServer.add(databaseServerId);
-                })
-                .subscribeOn(Schedulers.io());
-        databaseServerDeleteCompletable.await();
-        if (!failedToDeleteDatabaseServer.isEmpty()) {
-            LOGGER.error("Can't delete database server: {}", failedToDeleteDatabaseServer);
-            throw new CloudbreakServiceException("Can't delete database server: " + failedToDeleteDatabaseServer);
+    public Optional<String> deleteDatabaseServer(AzureClient azureClient, String databaseServerId, boolean cancelException) {
+        return handleDeleteErrors(resourceId -> {
+            LOGGER.info("Delete database server: {}", resourceId);
+            azureClient.deleteDatabaseServer(resourceId);
```

Comment:
This could be an async operation as we don't need to wait for it we can throw a CloudbreakServiceException inside doOnError()

Reply:
Subject candidate for future refactor